### DISCUSSION TITLE
About custom component

### DISCUSSION THREAD

**Discussion by momo91377** [web-infra-dev/rspress#1694] [Q&A] [answered]
I want to add an icon and a jump link to this location on each document page, do I need to customize this component in the theme?

**Comment by momo91377** [comment]
![image](https://github.com/user-attachments/assets/3f1a09c7-5a7b-4c0a-b535-2f3343e8f33b)

**Reply by Timeless0911** [reply]
refer to https://github.com/web-infra-dev/rslib/blob/ac486b96c93977227eba4b14518cdb11270ce0e1/website/docs/en/config/rsbuild/plugins.mdx?plain=1#L1-L3

**Reply by momo91377** [reply]
Didn't quite understand, does this apply in rspress as well?

**Reply by Timeless0911** [reply]
This is just an Rspress example.

**Reply by momo91377** [reply]
再请教一下，上述功能是否能通过自定义的bundle.js文件中对标题H1的写法来实现呢？

**Reply by Timeless0911** [reply, marked as answer]
![image](https://github.com/user-attachments/assets/93961d68-0c45-40b9-a77f-c73efbc2041a)
https://rspress.dev/guide/advanced/custom-theme#2-use-slot
You can use `components` to customize below MDX components yourself.

``` ts
{
	h1: H1,
	h2: H2,
	h3: H3,
	h4: H4,
	h5: H5,
	h6: H6,
	ul: Ul,
	ol: Ol,
	li: Li,
	table: Table,
	td: Td,
	th: Th,
	tr: Tr,
	hr: Hr,
	p: P,
	blockquote: Blockquote,
	strong: Strong,
	a: A,
	code: Code,
	pre: Pre,
	img: Img,
}
```
